app/controllers/workflows_controller.rb

```diff
@@ -35,11 +35,14 @@ def show
   end
 
   def update
-    if params['current-status'].present?
-      find_step_for_process.update(status: params['current-status'])
-    else
-      find_step_for_process.update(status: 'completed')
-    end
+    parser = ProcessParser.new(process_from_request_body)
+    step = find_step_for_process
+
+    return render plain: process_mismatch_error(parser), status: :bad_request if parser.process != params[:process]
+
+    return render plain: status_mismatch_error(step), status: :precondition_failed if params['current-status'] && step.status != params['current-status']
```

Comment:
This is probably the wrong status code. It doesn’t match the spec https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412

Reply:
Good call.

Reply:
@jcoyne How about 409 Conflict? 400 Bad Request? other?

Reply:
Went with 409 due to conflict in current status.